lib/TableGen/Constraints.cpp

```diff
@@ -312,7 +312,8 @@ StringRef MetaType::getGetterCppType() const {
     return "::llvm::Value *";
 
   if (isVarArgList())
-    return "::llvm::iterator_range<::llvm::Value *>";
+    return "::llvm::iterator_range<::llvm_dialects::const_op_to_value_iterator_"
```

Comment:
Can we just return llvm::iterator_range<llvm::Use *> here? LLVM already provide an adaptor iterator to make iterating over the Values easy. See here: https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/IR/User.h#L250.

Reply:
Thanks -- I thought there must be something like this somewhere but couldn't find it :)

I do think this should return an iteration over Values and not Uses. It's technically less powerful, but it's less surprising given the better symmetry between what you put into the builder vs. what you get from the getter. I'll look into simplifying the change using the `User::value_op_iterator` next week.